src/python/gudhi/wasserstein/wasserstein.py

```diff
@@ -88,32 +93,150 @@ def _perstot(X, order, internal_p, enable_autodiff):
         return np.linalg.norm(_dist_to_diag(X, internal_p), ord=order)
 
 
-def wasserstein_distance(X, Y, matching=False, order=1., internal_p=np.inf, enable_autodiff=False):
+def _get_essential_parts(a):
+    '''
+    :param a: (n x 2) numpy.array (point of a diagram)
+    :returns: five lists of indices (between 0 and len(a)) accounting for the five types of points with infinite
+    coordinates that can occur in a diagram, namely:
+        type0 : (-inf, finite)
+        type1 : (finite, +inf)
+        type2 : (-inf, +inf)
+        type3 : (-inf, -inf)
+        type4 : (+inf, +inf)
```

Comment:
Do you count those as being on the diagonal or infinitely far from the diagonal? It isn't well defined what to do with them, I think it would be ok not to support them, but if we support them (maybe we set some filtration values to infinity to "remove" some simplices but still computed the whole diagram), ignoring them (they are on the diagonal) seems more likely to be useful than considering them as really essential. Although returning inf or NaN may advertise that something strange is going on...

Reply:
I get your point. Right now I count them as essential part, i.e. infinitely away from the diagonal. 

One option would be to ignore them but still raising a warning, in the vein of `Warning: points with (+inf,+inf) and (-inf, -inf) are considered as belonging to the diagonal and have been removed.` ?

Reply:
https://github.com/grey-narn/hera/blob/b528c4067a8aac346eb307d3c23b82d5953cfe2d/wasserstein/include/wasserstein.h#L286
For reference, Hera now considers (+inf, +inf) as a diagonal point.

Reply:
Ok I will modify my code to behave in a consistent way (ie considering (+inf,+inf) and (-inf, -inf) as belonging to the diagonal).